[2025-10-18T21:50:01.203+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-18T21:50:01.223+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T21:45:00+00:00 [queued]>
[2025-10-18T21:50:01.230+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T21:45:00+00:00 [queued]>
[2025-10-18T21:50:01.231+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-10-18T21:50:01.241+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): fetch_weather_data> on 2025-10-18 21:45:00+00:00
[2025-10-18T21:50:01.248+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1361) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-10-18T21:50:01.249+0000] {standard_task_runner.py:63} INFO - Started process 1364 to run task
[2025-10-18T21:50:01.248+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'flood_prediction_pipeline', 'fetch_weather_data', 'scheduled__2025-10-18T21:45:00+00:00', '--job-id', '172', '--raw', '--subdir', 'DAGS_FOLDER/flood_prediction_dag.py', '--cfg-path', '/tmp/tmpj0ak00qs']
[2025-10-18T21:50:01.250+0000] {standard_task_runner.py:91} INFO - Job 172: Subtask fetch_weather_data
[2025-10-18T21:50:01.288+0000] {task_command.py:426} INFO - Running <TaskInstance: flood_prediction_pipeline.fetch_weather_data scheduled__2025-10-18T21:45:00+00:00 [running]> on host fcca755cbcb6
[2025-10-18T21:50:01.354+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='flood_prediction_pipeline' AIRFLOW_CTX_TASK_ID='fetch_weather_data' AIRFLOW_CTX_EXECUTION_DATE='2025-10-18T21:45:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-18T21:45:00+00:00'
[2025-10-18T21:50:01.355+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-18T21:50:01.388+0000] {logging_mixin.py:188} INFO - âœ… Weather data fetched: {'temp': 26.18703145356796, 'humidity': 62.04323720147006, 'pressure': 1020.073527580912, 'wind_speed': 1.5385285932709758, 'rain_1h': 6.835347046901177, 'rain_3h': 44.62906654785638, 'rain_6h': 128.09617233883182, 'rain_24h': 59.21339803949692, 'rain_intensity': 2.023313291668569, 'temp_delta': -3.0458933254866096, 'humidity_delta': -33.05970389569376, 'year': 2025, 'month': 10, 'day': 18, 'hour': 21, 'minute': 50, 'second': 1}
[2025-10-18T21:50:01.389+0000] {python.py:237} INFO - Done. Returned value was: {'temp': 26.18703145356796, 'humidity': 62.04323720147006, 'pressure': 1020.073527580912, 'wind_speed': 1.5385285932709758, 'rain_1h': 6.835347046901177, 'rain_3h': 44.62906654785638, 'rain_6h': 128.09617233883182, 'rain_24h': 59.21339803949692, 'rain_intensity': 2.023313291668569, 'temp_delta': -3.0458933254866096, 'humidity_delta': -33.05970389569376, 'year': 2025, 'month': 10, 'day': 18, 'hour': 21, 'minute': 50, 'second': 1}
[2025-10-18T21:50:01.390+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-18T21:50:01.406+0000] {taskinstance.py:1206} INFO - Marking task as SUCCESS. dag_id=flood_prediction_pipeline, task_id=fetch_weather_data, run_id=scheduled__2025-10-18T21:45:00+00:00, execution_date=20251018T214500, start_date=20251018T215001, end_date=20251018T215001
[2025-10-18T21:50:01.424+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2025-10-18T21:50:01.438+0000] {taskinstance.py:3498} INFO - 1 downstream tasks scheduled from follow-on schedule check
[2025-10-18T21:50:01.441+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
